ldapauthenticator/ldapauthenticator.py

```diff
@@ -58,6 +58,31 @@ def _server_port_default(self):
         """
     )
 
+    lookup_dn = Bool(
+        False,
+        config=True,
+        help='Look up the user\'s DN based on an attribute'
+    )
+
+    user_search_base = Unicode(
+        config=True,
+        help="""Base for looking up user accounts in the directory.
+
+        Example:
+
+            ou=people,dc=wikimedia,dc=org
+        """
+    )
+
+    user_attribute = Unicode(
+        config=True,
+        help="""LDAP attribute that stores the user's username.
+
+        For most LDAP servers, this is uid.  For Active Directory, it is
+        sAMAccountName.
```

Comment:
Nope, it's actually [sAMAccountName](https://msdn.microsoft.com/en-us/library/ms677605(v=vs.85).aspx) in keeping with the LDAP convention that attribute names always start lowercase.